### PR TITLE
Refactor, move solana-validator CLI related configs to its own module

### PR DESCRIPTION
**Pull request by eloylp**
#### Problem

The main function of the validator is getting big. I found it a bit difficult to track the code. 

#### Summary of Changes
There is  **no behavioural change** on this PR. Its just a refactor:

* Move the CLI initialisation code and validations to their own module, `cli.rs`, leaving the main function a bit more lightweight.
* Group all the default arguments in a dedicated struct. So easier to find and alter them.

#### Checks that i did
The most sensible change here are the defaults grouping. As the CLI defaults are shown in the output when invoking it with `--help`, i thought the following commands to be useful for a minimal check, to ensure no behavioural change:

```bash
cd solana/validator
## Save CLI output with defaults in master
$ git checkout master
$ cargo run -- --help > /tmp/master-cli-output.txt
## Save CLI output with defaults in the refactor branch
$ git checkout refactor-validator-main
$ cargo run -- --help > /tmp/refactor-cli-output.txt
## Compare md5 sums 
$ md5sum /tmp/master-cli-output.txt /tmp/refactor-cli-output.txt
12768cba6bb80d63e77b52fe162657d5  /tmp/master-cli-output.txt
12768cba6bb80d63e77b52fe162657d5  /tmp/refactor-cli-output.txt
```
And similar for the test validator part:

```bash
$ cd solana/validator

$ git checkout master
$ ./solana-test-validator --help > /tmp/master-test-cli-output.txt

$ git checkout refactor-validator-main
$ ./solana-test-validator --help > /tmp/refactor-test-cli-output.txt

$ md5sum /tmp/master-test-cli-output.txt /tmp/refactor-test-cli-output.txt 
c75c7b2a81c23961f68929838d07d5b9  /tmp/master-test-cli-output.txt
c75c7b2a81c23961f68929838d07d5b9  /tmp/refactor-test-cli-output.txt
```

Edit: updated with latest changes (conflict solving).

Edit2: updated with latest changes (conflict solving). Added test validator check.

### CODE REVIEW

**Comment by eloylp** [comment]
Solved conflicts. Looks like 2 arguments related to blockstore were deprecated at https://github.com/solana-labs/solana/pull/28409 . Ready for review !

**Comment by eloylp** [comment]
Hi @mvines , apologies for the direct ping, just trying to follow [contributing guides](https://github.com/solana-labs/solana/blob/master/CONTRIBUTING.md#getting-pull-requests-merged) . Could you help me on reviewing this ? Thank you in advance !

**Comment by eloylp** [comment]
Ouups, just realised it got conflicts again. Will move it to draft again. Early feedback still appreciated.

**Comment by eloylp** [comment]
The branch is now updated with latest changes. The last conflict was due to this [new addition](https://github.com/solana-labs/solana/pull/28707/files#diff-59e2068dd4d49c21ab87f3a5d8792d2daca826ab6924b32413e551c0e3035f61) . I think this is ready for review again.

File: validator/src/bin/solana-test-validator.rs [hunk 1]
```diff
@@ -246,7 +246,7 @@ fn main() {
             Arg::with_name("no_bpf_jit")
                 .long("no-bpf-jit")
                 .takes_value(false)
-                .help("Disable the just-in-time compiler and instead use the interpreter for SBF. Windows always disables JIT."),
+                .help("Disable the just-in-time compiler and instead use the interpreter for BPF. Windows always disables JIT."),
```
Comment (mvines) [inline validator/src/bin/solana-test-validator.rs:249]: Oops can you please restore this line

**Review by mvines — COMMENTED** [review]
Thanks for doing this. Looks fine to me overall, just one nit.

Do you think `validator/src/bin/solana-test-validator.rs` deserves the same treatment for its cli config?

**Comment by mvines** [comment]
Looks like CI wants you to run `cargo fmt` :)

**Comment by eloylp** [comment]
Hello @mvines. Here is a list of updates:

* > Do you think validator/src/bin/solana-test-validator.rs deserves the same treatment for its cli config?

Yes. I was worried about the size of the PR. Planned to do it a second PR. But here is it f31988c5899ba1f4484f52d29072b487daa07b45 . Let me know what do you think. I did the same md5 checks as in the first refactor (see updated PR description). 

* Updated with latest changes from master. A conflict was fixed, due to [this PR](https://github.com/solana-labs/solana/pull/28762/files#diff-59e2068dd4d49c21ab87f3a5d8792d2daca826ab6924b32413e551c0e3035f61) . I did again the md5 checks (see PR description).

* Configured `cargo fmt` on save for my new setup. It should not fail again :)

Probably a squash is a good idea if merged. I think this is ready for review again !

File: validator/src/bin/solana-test-validator.rs (hunk 1 above)
Comment (eloylp) [reply validator/src/bin/solana-test-validator.rs]: Fixed at ae6f93f303bd9c48db48b9488276c9a0006097c8

**Comment by eloylp** [comment]
Ah ! gotcha. CI failed again. Thats probably because:

```bash
$ cargo fmt --all
Warning: can't set `imports_granularity = One`, unstable features are only available in nightly channel.
...
```
This one did the trick:

```bash
cargo +nightly fmt --all
````

Fixed at 5b05edcbd4dbc3ac82599bd6f58a55117f9cc8bf . I hope. Sorry inconveniences.

**Comment by mvines** [comment]
Thanks for the PR!

**Comment by eloylp** [comment]
> Thanks for the PR!

Thanks for review 🙂

**Pull request merged**